### PR TITLE
support customize separator

### LINKED ISSUE #9891: RangePicker: allow the range separator to be overridden
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?
There is currently no way of overriding the range picker separator to set a custom value, e.g. 2018-01-01 to 2018-01-31, where " to " is an example of a custom value.  This would allow us to use wording that better reflects how we are using the RangePicker.

### What does the proposed API look like?
The `RangePicker` component accepts a new prop called `separator` which has a default value of `" ~ "` to match the current functionality.

### PR DESCRIPTION
**Pull request by zombieJ**
First of all, thank you for your contribution! 😄

New feature please send pull request to feature branch, and rest to master branch.
Pull request will be merged after one of collaborators approve.
Please makes sure that these form are filled before submitting your pull request, thank you!

[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Branch merge
- [ ] Other (about what?)

### What's the background?

close #9891

### API Realization (Optional if not new feature)

```jsx
<RangePicker separator="test" />
```

### What's the effect? (Optional if not new feature)

Popup style will change with input & separator line.

### Changelog description (Optional if not new feature)

> 1. DatePicker support `separator` prop.
> 2. DatePicker 支持 `separator` 属性。

### Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/date-picker/index.en-US.md [hunk 1]
```diff
@@ -122,6 +122,7 @@ The following APIs are shared by DatePicker, MonthPicker, RangePicker, WeekPicke
 | format | to set the date format | string | "YYYY-MM-DD HH:mm:ss" |
 | ranges | preseted ranges for quick selection | { \[range: string]: [moment](http://momentjs.com/)\[] } \| { \[range: string]: () => [moment](http://momentjs.com/)\[] } | - |
 | renderExtraFooter | render extra footer in panel | () => React.ReactNode | - |
+| separator | (Add in 3.14.0) set separator between inputs | string | '~' |
```
Comment (afc163) [inline components/date-picker/index.en-US.md]: 版本信息加一列吧。https://github.com/ant-design/ant-design/issues/14818

Comment (zombieJ) [reply components/date-picker/index.en-US.md]: 加的一列叫 `Version` | `版本`以节约列宽度。

**Comment by afc163** [comment]
image

`~` 的位置和之前的不太一样了。

**Comment by zombieJ** [comment]
> `~` 的位置和之前的不太一样了。

调了一下，再看看~

**Review by afc163 — APPROVED** [review]

**Comment by Brlja** [comment]
typescript prop does not work.

**Comment by ztplz** [comment]
@Brlja More detail information?

**Comment by Brlja** [comment]
I get an error that property "separator" does not exist. Using the latest version. Twice checked.

**Pull request merged**